lib/client-assets.php

```diff
@@ -299,7 +299,7 @@ function gutenberg_register_scripts_and_styles() {
 	gutenberg_override_style(
 		'wp-block-library',
 		gutenberg_url( 'build/block-library/style.css' ),
-		current_theme_supports( 'wp-block-styles' ) ? array( 'wp-block-library-theme' ) : array(),
```

Comment:
I'm not understanding the reason for this change. I think 'wp-block-styles' should still depend on 'wp-block-library-theme' if the theme supports them.
gutenberg_override_style calls wp_register_style with the new dependencies so even if this dependency was set in core as we are registering it again we still need to pass it.

Reply:
I have vague memories that it's done in a different way in Core and this might align both approaches. The thing is, it's not possible to call `current_theme_supports` in the script loader in Core.

Reply:
Yes, there's a bit more of an explanation in in the extended comment description for https://github.com/WordPress/gutenberg/pull/13569/commits/2282ee4143db1d2f6ee01327317813446a4b2a8e . It still exists, but it's manually enqueued in core's implementation of the block scripts and styles function:

https://github.com/WordPress/wordpress-develop/blob/e421f26/src/wp-includes/script-loader.php#L2626-L2630

You can confirm this adding the following snippet somewhere in a theme which adds support for `wp-block-styles` (e.g. Twentynineteen):

```php
add_action( 'enqueue_block_assets', function() {
	var_export( current_theme_supports( 'wp-block-styles' ) );
	var_export( wp_style_is( 'wp-block-library-theme', 'enqueued' ) );
	exit;
} );
```

Should display "truetrue" on the editor screen.

Reply:
Thank you for the clarification @youknowriad, and @aduth.